### PR TITLE
Fix bug for nco we2e tests on Orion; re-organize we2e input data and nco we2e tests

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 
- Re-organize the input data for the WE2E tests on `Orion` and `WCOSS_dell_p3` to match up with those on `Hera`.
- Fill in the missing configuration for `Orion` in the 'ush/machine' directory to make the 'nco' WE2E tests work on `Orion`.
- Minimize the number of the WE2E tests for the 'nco' mode since all the major features are tested on the 'community' mode.

## TESTS CONDUCTED: 
1. WE2E tests on `Orion`, `WCOSS_dell_p3`, and `Hera`:
- community_ensemble_2mems
- grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
- grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
- grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
- grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
- grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
- grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
- grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_RRFS_v1beta
- grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1alpha
- grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km
- nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR
- nco_grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
- nco_grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km
- nco_ensemble

2. Extra WE2E tests on `WCOSS_dell_p3` and `Hera`:
- MET_ensemble_verification
- MET_verification

### CODE REVIEW

**Review by jwolff-ncar — APPROVED** [review]
I don't have access to Orion to test this but visual it looks good to me.

**Comment by mkavulich** [comment]
@chan-hoo Can you put a bit more detail about these changes to the WE2E tests? I understand the test deletions for reducing the number of NCO tests, but in others you have changed physics options and boundary condition settings and I'm interested in why those were changed.

File: tests/WE2E/run_WE2E_tests.sh [hunk 1]
```diff
@@ -941,7 +941,8 @@ machine (MACHINE):
   MACHINE= \"${MACHINE}\""
     fi
     EXTRN_MDL_SOURCE_BASEDIR_ICS="${extrn_mdl_source_basedir}/${EXTRN_MDL_NAME_ICS}"
-    if [ "${EXTRN_MDL_NAME_ICS}" = "FV3GFS" ] && [ "$MACHINE" = "HERA" ]; then
+    if [[ "${EXTRN_MDL_NAME_ICS}" = "FV3GFS" &&
+      ( "$MACHINE" = "HERA" || "$MACHINE" = "WCOSS_DELL_P3" || "$MACHINE" = "ORION" ) ]]; then
```
Comment (christinaholtNOAA) [inline tests/WE2E/run_WE2E_tests.sh]: Can the platform be removed from this logic altogether after @mkavulich recently synced data across platforms?

Comment (chan-hoo) [reply tests/WE2E/run_WE2E_tests.sh]: @christinaholtNOAA, I only have access to Hera, Orion, and WCOSS_dell_p3. If you confirm that the other platforms have the same data structure, I'll remove this line.

**Comment by chan-hoo** [comment]
@mkavulich, we'd like to test three domains [RRFS_CONUS_25km (for test), RRFS_CONUS_13km (for RRFS-CMAQ), RRFS_CONUS_3km (for parallel)], and three ccpp suites [FV3_GFS_v15_thompson_mynn_lam3km (major interest at EMC), FV3_GFS_v15p2 and _v16 (officially supported suites in the UFS SRW App as well as in RRFS-CMAQ)] in the 'nco' mode. Since the '*_thompson_mynn_lam3km' should be used with the 3km resolution domain, I matched it with the RRFS_CONUS_3km domain. In addition, the 'inline post' option is widely used at EMC. As you know, the difference between the 'community' and 'nco' modes is the structure of the input and output data. I think the above tests are good enough for the 'nco' mode at this moment. For IC/BCs, I've set the same date, 20190615, for all the tests for convenience. In the staged data set, the specific date 20190615 provides grib2 files every 6 hours. This is the reason why I changed 'LBC_SPEC_INTVL_HRS' from '3' to '6'. Finally, I'd like to test one case for two BC intervals (12 hours). I set this to the lowest resolution domain (25km) to save its computing time.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16.sh [hunk 2]
```diff
@@ -3,24 +3,25 @@
...
 FV3GFS_FILE_FMT_LBCS="grib2"
+USE_USER_STAGED_EXTRN_FILES="TRUE"
 
 DATE_FIRST_CYCL="20190615"
 DATE_LAST_CYCL="20190615"
 CYCL_HRS=( "00" )
 
-FCST_LEN_HRS="6"
-LBC_SPEC_INTVL_HRS="3"
+FCST_LEN_HRS="12"
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16.sh]: @chan-hoo Actually, the data is available every 3 hours in the original directory on Hera at `/scratch2/BMC/det/Gerard.Ketefian/UFS_CAM/staged_extrn_mdl_files/FV3GFS/grib2/2019061500`.  I think @mkavulich copied it over to the official location at `/scratch2/BMC/det/UFS_SRW_app/develop/model_data/FV3GFS/grib2/2019061500`.  Not sure why only 6-hour output was copied.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh [hunk 3]
```diff
@@ -1,25 +0,0 @@
-#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh]: I'm pretty sure we want to keep this because the GSL group still runs the HRRR suite in NCO mode with HRRR ICs and RAP LBCs.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_inline_post.sh [hunk 4]
```diff
@@ -10,19 +10,22 @@ RUN_ENVIR="nco"
...
 
-DATE_FIRST_CYCL="20190701"
-DATE_LAST_CYCL="20190701"
+DATE_FIRST_CYCL="20190615"
+DATE_LAST_CYCL="20190615"
 CYCL_HRS=( "00" )
 
 FCST_LEN_HRS="6"
-LBC_SPEC_INTVL_HRS="3"
+LBC_SPEC_INTVL_HRS="6"
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_inline_post.sh]: @chan-hoo The 3-hourly data for this is actually available, just needs to be copied over to the proper destination (on Orion, Hera, etc).

File: tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh [hunk 5]
```diff
@@ -1,35 +0,0 @@
-#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh]: @chan-hoo After discussing this during the DTC SRW App project meeting, I think we want to keep this as well since the code that runs the combination of ensembles and NCO mode is only tested here.  I'm pretty sure eventually this will be used by GSL (and probably also EMC, no?).

Comment (chan-hoo) [reply tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh]: @gsketefian, OK. I'll restore them (ensemble, hrrr). Instead, I'll remove the FV3GFS conus 25km case because the 25km domain and FV3GFS are being tested in other cases. Should I copy the missing 3 hour data to the central location?

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_inline_post.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# 
+#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_inline_post.sh]: @chan-hoo The purpose of this test is to check whether the inline post works in NCO mode, so it is a test of a workflow feature (or a combination of two features being enabled, inline post and NCO mode).  That's why it's in the `wflow_features` directory.  And that's why it's a short forecast (3 hours), since if it works for the first 4 outputs, it's probably going to work for all hours.  The directory `grids_extrn_mdls_suites_nco` is more for scientific tests, i.e. to check that the important combinations of grids, external models, and suites work.  

If you want to reduce the number of tests, one thing you can do is make one of the other tests in `grids_extrn_mdls_suites_nco` use `WRITE_POST="TRUE"` and replace this file with a symlink that points to that test.  The symlink can have the same name as this file, i.e.
`wflow_features/config.nco_inline_post.sh -->  ../grids_extrn_mdls_suites_nco/config.whichever_test_you_want_to_enable_inline_post_in.sh`
The testing script (`run_WE2E_tests.sh`) knows what to do with symlinks like this.

Comment (chan-hoo) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_inline_post.sh]: @gsketefian, agreed. I'll remove this test and put the inline post option into one of the existing test cases.

**Comment by chan-hoo** [comment]
@mkavulich, @gsketefian, Can you add the missing 3 hour data to the central location on Hera (/scratch2/BMC/det/UFS_SRW_app/develop/model_data/FV3GFS/grib2/2019061500)? I don't have permission for this.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh (hunk 3 above)
Comment (chan-hoo) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh]: returned.

File: tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh (hunk 5 above)
Comment (chan-hoo) [reply tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh]: returned.

**Comment by chan-hoo** [comment]
@gsketefian, I copied the missing data from your original directory on Hera to Orion and WCOSS_dell_p3. The new nco tests worked well on Orion and WCOSS_dell_p3. They should work if the missing data is copied to the designated directory on Hera. If you have any other comments, let me know.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16.sh (hunk 2 above)
Comment (mkavulich) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16.sh]: Gerard and I talked offline, but for the benefit of everyone else: We are trying to minimize the size of static data being maintained, so I only copied over data that was being used in tests actively. The test as it is on Hera right now does not use static data, it pulls it from HPSS.

**Comment by mkavulich** [comment]
@chan-hoo the 3-hour data is now available on Hera at /scratch2/BMC/det/UFS_SRW_app/develop/model_data/FV3GFS/grib2/2019061500

File: tests/WE2E/run_WE2E_tests.sh (hunk 1 above)
Comment (mkavulich) [reply tests/WE2E/run_WE2E_tests.sh]: We should converge on a single solution for how the data is staged, there doesn't seem to be any reason to maintain two different structures on different platforms. I asked @JeffBeck-NOAA and @gsketefian to weigh in on which we should move to.

Comment (mkavulich) [reply tests/WE2E/run_WE2E_tests.sh]: @chan-hoo you can remove the machine-based logic from that statement; I will move the data on the various platforms to conform to the Hera (and now WCOSS) directory structure.

Comment (chan-hoo) [reply tests/WE2E/run_WE2E_tests.sh]: removed

File: tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh [hunk 7]
```diff
@@ -1,28 +0,0 @@
-#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh:1]: Which under `grids_extrn_mdls_suites_nco` is now the test for inline post?

File: tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh [hunk 8]
```diff
@@ -17,16 +17,16 @@
 RUN_ENVIR="nco"
 PREEXISTING_DIR_METHOD="rename"
 
-PREDEF_GRID_NAME="CONUS_25km_GFDLgrid"
-CCPP_PHYS_SUITE="FV3_GFS_2017_gfdlmp_regional"
+PREDEF_GRID_NAME="RRFS_CONUS_25km"
+CCPP_PHYS_SUITE="FV3_GFS_v15p2"
```
Comment (gsketefian) [inline tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh:21]: @chan-hoo Just curious why you're using 15p2 here since you removed the other 15p2 tests above.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh [hunk 9]
```diff
@@ -17,9 +17,11 @@ EXTRN_MDL_NAME_ICS="HRRR"
 EXTRN_MDL_NAME_LBCS="RAP"
 USE_USER_STAGED_EXTRN_FILES="TRUE"
 
-DATE_FIRST_CYCL="20200208"
-DATE_LAST_CYCL="20200208"
+DATE_FIRST_CYCL="20200810"
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh:20]: @chan-hoo How are you deciding on new dates?

File: tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh (hunk 7 above)
Comment (gsketefian) [reply tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh:1]: Oh sorry, I just saw it's `config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh`.  Can you add a symlink here (named `config.nco_inline_post.sh`) to that file so we know where that is?

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh (hunk 9 above)
Comment (chan-hoo) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh:20]: @gsketefian, I matched the date with the same test in the 'community' mode (config.grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh).

File: tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh (hunk 8 above)
Comment (chan-hoo) [reply tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh:21]: @gsketefian, there was no special intention. v15p2 was officially supported in release v1 and it will be used for RRFS-CMAQ along with v16. I thought v15p2 would be better than 2017_gfdlmp_regional. As you said, v15p2 was not tested in other nco we2e tests. So I applied it to this test.

File: tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh (hunk 7 above)
Comment (chan-hoo) [reply tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh:1]: done.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh (hunk 9 above)
Comment (gsketefian) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_nco/config.nco_grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR.sh:20]: Ok, makes sense.

File: tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh (hunk 8 above)
Comment (gsketefian) [reply tests/WE2E/test_configs/wflow_features/config.nco_ensemble.sh:21]: Ok, sounds good.

File: tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh (hunk 7 above)
Comment (gsketefian) [reply tests/WE2E/test_configs/wflow_features/config.nco_inline_post.sh:1]: Thanks.

**Review by gsketefian — APPROVED** [review]
Looks good.
